src/autometrics/decorator.py

```diff
@@ -29,49 +37,103 @@ def autometrics(
     *,
     objective: Optional[Objective] = None,
 ):
+    def track_result_ok(start_time: float, function: str, module: str, caller: str):
+        get_tracker().finish(
+            start_time,
+            function=function,
+            module=module,
+            caller=caller,
+            objective=objective,
+            result=Result.OK,
+        )
+
+    def track_result_error(
+        exception: Exception,
+        start_time: float,
+        function: str,
+        module: str,
+        caller: str,
+    ):
+        get_tracker().finish(
+            start_time,
+            function=function,
+            module=module,
+            caller=caller,
+            objective=objective,
+            result=Result.ERROR,
+        )
+        # Reraise exception
+        raise exception
+
     """Decorator for tracking function calls and duration."""
 
     def decorator(func: Callable[P, T]) -> Callable[P, T]:
         module_name = get_module_name(func)
```

Comment:
this logic might not need to live here too — i could try factoring out of `deocrator` and `async_decorator`